pyproject.toml

```diff
@@ -87,7 +89,8 @@ docs = [
     "sphinx-autobuild",
     "sphinx-autodoc-typehints",
     "sphinx-lint",
-    "sphinx-sitemap>=2.6.0",
+    # sphinx-sitemap>=2.7 fails to build on RTD infra, likely due to https://github.com/jdillard/sphinx-sitemap/pull/95
+    "sphinx-sitemap~=2.6.0",
```

Comment:
@medubelko, this release came out less than 24 hours after you landed #5559 :facepalm: 

I don't want to block this PR, but we may need to debug this in the future.

The error is:
```
sphinx-sitemap: No pages generated for sitemap.xml

Warning, treated as error:
Error getting data from Git (no "last updated" dates will be shown for source files from /home/docs/checkouts/readthedocs.org/user_builds/canonical-snapcraft/envs/5577/lib/python3.12/site-packages/craft_parts_docs/craft-parts/explanation):
fatal: not a git repository (or any parent up to mount point /home/docs/checkouts/readthedocs.org/user_builds)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).

make: *** [common.mk:225: docs] Error 2
```

Reply:
Thanks. My first guess is that we can add craft-parts' common docs to `git_exclude_patterns` ([source](https://github.com/mgeier/sphinx-last-updated-by-git/blob/master/README.rst)).